src/background/actions/TokenSigning/sign.ts

```diff
@@ -53,18 +75,54 @@ export default async function sign(
   const nativeAppService = new NativeAppService();
 
   try {
+    const warnings: Array<string> = [];
     const nativeAppStatus = await nativeAppService.connect();
 
     config.DEBUG && console.log("Sign: connected to native", nativeAppStatus);
 
+    let hashFunction = (
+      Object.keys(digestCommandToHashFunction).includes(algorithm)
+        ? digestCommandToHashFunction[algorithm]
+        : algorithm
+    );
+
+    const expectedHashByteLength = (
+      Object.keys(hashFunctionToLength).includes(hashFunction)
+        ? hashFunctionToLength[hashFunction]
+        : undefined
+    );
+
+    const hashByteArray = new ByteArray().fromHex(hash);
+
+    if (hashByteArray.length !== expectedHashByteLength) {
+      warnings.push(
+        `${algorithm} hash must be ${expectedHashByteLength} bytes long.\n` +
+        `The provided hash was ${hashByteArray.length} bytes long.\n` +
+        "See further details at https://github.com/web-eid/web-eid-webextension#hwcrypto-compatibility"
+      );
+
+      const autodetectedHashFunction = Object.keys(hashFunctionToLength).find((hashFunctionName) => (
+        hashFunctionToLength[hashFunctionName] == hashByteArray.length)
+      );
```

Comment:
`===` is always preferable AFAIK and a possible minor aesthetics improvement with the closing parenthesis: 
```suggestion
        hashFunctionToLength[hashFunctionName] === hashByteArray.length
      ));
```